### PR TITLE
Issue/338 – Batch importer architecture (WIP)

### PR DESCRIPTION
**Pull request by DrewAPicture**
Issue: #338

This stubs out a PR for introducing batch import architecture. It is built to extend functionality introduced in v2.0 and will be tackled at a later date.

### CODE REVIEW

File: includes/admin/ajax-actions.php [hunk 1]
```diff
@@ -139,7 +139,18 @@ function affwp_process_batch_request() {
 	 *
 	 * @var \AffWP\Utils\Batch_Process\Export|\AffWP\Utils\Batch_Process\Base $process
 	 */
-	$process = new $class( $step );
+	if ( isset( $_REQUEST['data']['upload']['file'] ) ) {
+
+		// If this is an import, instantiate with the file and step.
+		$file = sanitize_text_field( $_REQUEST['data']['upload']['file'] );
+		$process = new $class( $file, $step );
```
Comment (pippinsplugins) [inline includes/admin/ajax-actions.php]: Could we have a class_exists sanity check here?

File: includes/admin/tools/import/class-batch-import-affiliates.php [hunk 2]
```diff
@@ -89,4 +192,64 @@ public function get_message( $code ) {
...
+				$first_name = sanitize_text_field( $args['first_name'] );
+			}
+
+			if ( ! empty( $args['last_name'] ) ) {
+				$last_name = sanitize_text_field( $args['last_name'] );
+			}
+
+			$user_id = wp_insert_user( array(
+				'user_login' => sanitize_user( $args['payment_email'], $this->use_strict ),
+				'user_email' => sanitize_email( $args['payment_email'] ),
```
Comment (pippinsplugins) [inline includes/admin/tools/import/class-batch-import-affiliates.php]: Let's not use sanitize_email() here as it strips out some characters, like `+` that are valid characters. Use `sanitize_text_field()` instead. See https://core.trac.wordpress.org/ticket/21537

File: includes/admin/ajax-actions.php (hunk 1 above)
Comment (DrewAPicture) [inline includes/admin/ajax-actions.php]: There's a `class_exists()` check on `$class` a few lines up on L#126.

File: includes/admin/tools/import/class-batch-import-affiliates.php (hunk 2 above)
Comment (DrewAPicture) [inline includes/admin/tools/import/class-batch-import-affiliates.php]: Roger.

**Pull request merged**